api/api-pipeline-instance-v1/src/main/java/com/thoughtworks/go/apiv1/pipelineinstance/PipelineInstanceControllerV1.java

```diff
@@ -68,12 +69,15 @@ public void setupRoutes() {
 
             before(Routes.PipelineInstance.INSTANCE_PATH, mimeType, apiAuthenticationHelper::checkPipelineViewPermissionsAnd403);
             before(Routes.PipelineInstance.HISTORY_PATH, mimeType, apiAuthenticationHelper::checkPipelineGroupOperateOfPipelineOrGroupInURLUserAnd403);
+            before(Routes.PipelineInstance.COMMENT_PATH, mimeType, apiAuthenticationHelper::checkPipelineGroupOperateOfPipelineOrGroupInURLUserAnd403);
```

Comment:
Do we need `operate` permissions for commenting.  Can't `view` be enough for this?
cc:\ @maheshp @adityasood @arvindsv

Reply:
@kritika-singh3 - I have kept the permission to operate as it is the same in the service. I read https://github.com/gocd/gocd/pull/699 but I was not able to figure out why we decided to go with `operate` permission

Reply:
I think we should retain the behavior as is and allow only users with operate permission to comment.

Reply:
ok.